From: Lubomir Rintel <lkundrak@v3.sk>
Date: Fri, 2 Jun 2017 13:09:07 +0200
Subject: clients: respond to the secret requests that we can't service

Otherwise the daemon would hang waiting for us while we respond with awkward
silence. That is not a healthy kind of communication.
---
 clients/common/nm-secret-agent-simple.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/clients/common/nm-secret-agent-simple.c b/clients/common/nm-secret-agent-simple.c
index 89cf156..97bbf0d 100644
--- a/clients/common/nm-secret-agent-simple.c
+++ b/clients/common/nm-secret-agent-simple.c
@@ -576,6 +576,15 @@ request_secrets_from_ui (NMSecretAgentSimpleRequest *request)
 		ok = FALSE;
 
 	if (!ok) {
+		gs_free_error GError *error = NULL;
+
+		error = g_error_new (NM_SECRET_AGENT_ERROR, NM_SECRET_AGENT_ERROR_FAILED,
+		                     "Cannot service a secrets request %s for a %s connection",
+		                     request->request_id,
+		                     nm_connection_get_connection_type (request->connection));
+		request->callback (NM_SECRET_AGENT_OLD (request->self), request->connection,
+		                   NULL, error, request->callback_data);
+		g_hash_table_remove (priv->requests, request->request_id);
 		g_ptr_array_unref (secrets);
 		return;
 	}
